include/DDML/DDMLEventAction.h

```diff
@@ -0,0 +1,70 @@
+#include <G4Types.hh>            // for G4int, G4double
+#include <vector>                // for vector
+#include "G4UserEventAction.hh"  // for G4UserEventAction
+//#include "G4Timer.hh"            // for G4Timer
+class G4Event;
+#include "DDG4/Geant4Handle.h"
+#include "DDG4/Geant4Kernel.h"
+#include "DDG4/Geant4EventAction.h"   
+#include "CLHEP/Units/SystemOfUnits.h"
+#include "CLHEP/Units/PhysicalConstants.h"
+
+
+namespace ddml {
+
+/** Event action for ddml. 
+*
+*  @author P. McKeown, CERN
+*  @date Feb 2025
+*
+*/
+
+class DDMLEventAction: public dd4hep::sim::Geant4EventAction{
+public:
+    /// Standard constructor with initializing arguments
+    DDMLEventAction(dd4hep::sim::Geant4Context* c, const std::string& n);
+    /// Default destructor
+    virtual ~DDMLEventAction();
+    /// begin-of-event callback
+    inline virtual void begin(const G4Event*) override;
+    /// End-of-event callback
+    virtual void end(const G4Event*) override;
```

Comment:
```suggestion
    /// End-of-event callback
    void end(const G4Event*) override;
```
`override` implies `virtual`